pkg/gofr/gofr.go

```diff
@@ -309,87 +311,101 @@ func (a *App) initTracer() {
 
 	traceExporter := a.Config.Get("TRACE_EXPORTER")
 	tracerURL := a.Config.Get("TRACER_URL")
+	authHeader := a.Config.Get("TRACER_AUTH_KEY")
 
-	// deprecated : tracer_host and tracer_port is deprecated and will be removed in upcoming versions
+	// deprecated : tracer_host and tracer_port are deprecated and will be removed in upcoming versions.
 	tracerHost := a.Config.Get("TRACER_HOST")
-	tracerPort := a.Config.GetOrDefault("TRACER_PORT", "9411")
+	tracerPort := a.Config.Get("TRACER_PORT")
 
-	if tracerURL != "" && traceExporter == "" {
-		a.Logger().Error("missing TRACE_EXPORTER config, should be provided with TRACER_URL to enable tracing")
-		return
+	if tracerURL == "" && (tracerHost != "" || tracerPort != "") {
+		a.Logger().Warn("TRACER_HOST and TRACER_PORT are deprecated, use TRACER_URL instead")
 	}
 
-	//nolint:revive // early-return is not possible here, as below is the intentional logging flow
-	if tracerURL == "" && traceExporter != "" && !strings.EqualFold(traceExporter, "gofr") {
-		if tracerHost != "" && tracerPort != "" {
-			a.Logger().Warn("TRACER_HOST and TRACER_PORT are deprecated, use TRACER_URL instead")
-		} else {
-			a.Logger().Error("missing TRACER_URL config, should be provided with TRACE_EXPORTER to enable tracing")
-			return
-		}
+	exporter, err := a.getExporter(traceExporter, tracerHost, tracerPort, tracerURL, authHeader)
+	if err != nil {
+		a.container.Error(err)
 	}
 
-	if (traceExporter != "" && tracerHost != "") || tracerURL != "" || traceExporter == gofrTraceExporter {
-		exporter, err := a.getExporter(traceExporter, tracerHost, tracerPort, tracerURL)
-
-		if err != nil {
-			a.container.Error(err)
-		}
-
-		batcher := sdktrace.NewBatchSpanProcessor(exporter)
-		tp.RegisterSpanProcessor(batcher)
-	}
+	batcher := sdktrace.NewBatchSpanProcessor(exporter)
+	tp.RegisterSpanProcessor(batcher)
 }
 
-func (a *App) getExporter(name, host, port, url string) (sdktrace.SpanExporter, error) {
+func (a *App) getExporter(name, host, port, url, authHeader string) (sdktrace.SpanExporter, error) {
 	var (
 		exporter sdktrace.SpanExporter
-		err      error
+		ctx      context.Context
```

Comment:
I'm a bit worried this context is defined by it's type, but never gets a value, so right now it's equal to nil unless I'm wrong

But this one is used by open telemetry

So maybe instead of setting here to context.Background (one solution)

The ctx could be passed as an argument of getExporter

Reply:
This was finally called at

```go
// Start establishes a gRPC connection to the collector.
func (c *client) Start(context.Context) error {
	if c.conn == nil {
		// If the caller did not provide a ClientConn when the client was
		// created, create one using the configuration they did provide.
		conn, err := grpc.NewClient(c.endpoint, c.dialOpts...)
		if err != nil {
			return err
		}
		// Keep track that we own the lifecycle of this conn and need to close
		// it on Shutdown.
		c.ourConn = true
		c.conn = conn
	}

	// The otlptrace.Client interface states this method is called just once,
	// so no need to check if already started.
	c.tscMu.Lock()
	c.tsc = coltracepb.NewTraceServiceClient(c.conn)
	c.tscMu.Unlock()

	return nil
}
```

Reply:
I  add the context.Background() as the argument of getExporter